src/main/java/org/dataone/indexer/storage/Storage.java

```diff
@@ -4,17 +4,74 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.security.NoSuchAlgorithmException;
+import java.util.Properties;
 
-import org.dataone.hashstore.exceptions.PidRefsFileExistsException;
-
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.dataone.configuration.Settings;
+import org.dataone.hashstore.HashStore;
+import org.dataone.hashstore.HashStoreFactory;
+import org.dataone.hashstore.exceptions.HashStoreFactoryException;
 
 /**
- * The Storage represents the interface to access the objects and system metadata
+ * The HashFileStore implementation of the Storage interface 
  */
-public interface Storage {
+public class Storage {
+
+    private static Log log = LogFactory.getLog(Storage.class);
+    private static Storage instance;
+    private static HashStore hashStore;
+    static {
+        try {
+            instance = new Storage();
+        } catch (IOException e) {
+            log.error(
```

Comment:
If an exception is thrown, it's critical and un-recoverable. I think we should add the exception `e` to the logging statement, too, so it logs the stacktrace. What do you think? should we even do a System.exit(), [like we do here](https://github.com/DataONEorg/dataone-indexer/blob/ea064aacb72a764e7d2720e13177c568566fcdba/src/main/java/org/dataone/cn/indexer/IndexWorker.java#L107)?